observatory-platform/observatory/platform/utils/gc_utils.py

```diff
@@ -1232,3 +1234,136 @@ def delete_bucket_dir(*, bucket_name: str, prefix: str):
 
     for blob in blobs:
         blob.delete()
+
+
+def list_all_buckets():
+    """List of all Google Cloud buckets.
+
+    :return: list of buckets that the service account as access to under the project.
+    """
+
+    # Client to read Google Cloud Storage
+    storage_client = storage.Client()
+    buckets = storage_client.list_buckets()
+
+    # Make list of all names of the storage buckets
+    bucket_list = []
+    for bucket in buckets:
+        bucket_list.append(bucket.name)
+
+    return bucket_list
```

Comment:
This function and `get_age_of_bucket_in_days` could be consolidated by directly returning the list of `Bucket` instances, since `list_buckets` already returns Bucket instances, and calling `get_age_of_bucket_in_days` effectively means that they are fetched twice.

Assuming that `list_buckets` is a generator, the code from 1247 to the end of the function could be replace by `list(storage_client.list_buckets())`. You can then perform the time difference calculation, calling `bucket.time_created` from within `delete_old_buckets_with_prefix`.

It would also be good to specify the return type of the function:
```
def list_all_buckets() -> List[Bucket]:
```

Bucket is the type `google.cloud.storage.bucket.Bucket`.